### PR TITLE
Remote logging test -- and partial fixes.

### LINKED ISSUE #219: On Pilot Logging
There are a few issues with the current implementation of PilotLogging. 

1) **SOLVED** activating *RemoteLogger* in an environment that does NOT include env `X509_CERT_DIR` and `X509_USER_PROXY` generates the following:

```
2023-10-31T15:48:30.341093Z DEBUG [PilotParams] Release project: 
2023-10-[31](https://github.com/fstagni/Pilot/actions/runs/6709010578/job/18231106705#step:6:32)T15:48:30.342023Z INFO [Pilot] Remote logger activated
Traceback (most recent call last):
  File "/home/runner/work/Pilot/Pilot/Pilot/dirac-pilot.py", line 76, in <module>
    log.buffer.flush()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 514, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 599, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 6[35](https://github.com/fstagni/Pilot/actions/runs/6709010578/job/18231106705#step:6:36), in sendMessage
    context.load_cert_chain(cert)
TypeError: certfile should be a valid filesystem path
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 522, in run
    self.function(*self.args, **self.kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 514, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 599, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 635, in sendMessage
    context.load_cert_chain(cert)
TypeError: certfile should be a valid filesystem path
```

This is especially true for the case when DIRAC is sourced from CVMFS (not installed).

To address this issue, in https://github.com/DIRACGrid/Pilot/pull/218 I moved the setting of these variables as early as possible. 

2) **NOT SOLVED** Testing the above PR (#218) when DIRAC is sourced from CVMFS, and we're using certificates and not proxies generates the following (https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881):

```
2023-11-01T11:50:36.304812Z DEBUG [PilotParams] Release project: 
2023-11-01T11:50:44.286345Z DEBUG [PilotParams] X509_CERT_DIR is set in the host environment as /cvmfs/grid.cern.ch/etc/grid-security/certificates, aligning installEnv to it
2023-11-01T11:50:44.287539Z DEBUG [PilotParams] X509_VOMS_DIR is set in the host environment as /cvmfs/grid.cern.ch/etc/grid-security/vomsdir, aligning installEnv to it
2023-11-01T11:50:44.287626Z DEBUG [PilotParams] X509_VOMSES is not set in the host environment
2023-11-01T11:50:44.287688Z DEBUG [PilotParams] Candidate directory for X509_VOMSES is /cvmfs/grid.cern.ch/etc/grid-security/vomses
2023-11-01T11:50:44.289156Z DEBUG [PilotParams] Setting X509_VOMSES=/cvmfs/grid.cern.ch/etc/grid-security/vomses
2023-11-01T11:50:44.289448Z INFO [Pilot] Remote logger activated
Traceback (most recent call last):
  File "/home/runner/work/Pilot/Pilot/Pilot/dirac-pilot.py", line 76, in <module>
    log.buffer.flush()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 6[31](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:32), in sendMessage
    res = urlopen(url, data, context=context)
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 6[34](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:35), in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: Unauthorized
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 511, in run
    self.function(*self.args, **self.kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 631, in sendMessage
    res = urlopen(url, data, context=context)
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error [40](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:41)1: Unauthorized
Error: Process completed with exit code 1.
```

this is because the host certificate is not recognized as such. A DIRAC client would add a `{""extraCredentials": "hosts"}` to the credential dictionary, but this is not done when running before DIRAC is installed.

3) **NOT SOLVED**. Testing the above PR (#218) when DIRAC is installed locally, and we're using certificates and not proxies generates the following (https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260566428) : 

```
2023-11-01T11:51:13.939924Z INFO [Pilot] Remote logger activated
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 511, in run
    self.function(*self.args, **self.kwargs)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 623, in sendMessage
    context.load_verify_locations(capath=caPath)
TypeError: cafile, capath and cadata cannot be all omitted
```

Again, this is because `X509_CERT_DIR` and `X509_USER_PROXY` are not defined at this stage. 

---

I don't know how to solve 2) and 3). 

Maybe for 2) the simplest would be to use a `diracx` service.

### PR DESCRIPTION
**Pull request by fstagni**
closes #219

### CODE REVIEW

File: Pilot/pilotTools.py [hunk 1]
```diff
@@ -849,7 +855,7 @@ def __init__(self):
         self.pilotCFGFile = "pilot.json"
         self.pilotLogging = False
         self.loggerURL = None
-        self.loggerTimerInterval = 600
+        self.loggerTimerInterval = 0
```
Comment (fstagni) [inline Pilot/pilotTools.py:858]: @martynia and @sfayer double check this

**Comment by fstagni** [comment]
@martynia and @atsareg final review of this? It will be merged anyway before tomorrow's hackathon.

**Comment by martynia** [comment]
Have you run your integration tests with RemoteLogger enabled ? I cannot see any log files on the server.

**Comment by fstagni** [comment]
> Have you run your integration tests with RemoteLogger enabled ? I cannot see any log files on the server.

Yes. Let's see later what's wrong, but maybe this time is on the server side. 
I merge for the hackathon.

**Comment by martynia** [comment]
Fine.

**Pull request merged**